tests/unit/utils/checksum.spec.ts

```diff
@@ -0,0 +1,406 @@
+import { constants, createReadStream } from 'node:fs';
+import { Readable } from 'node:stream';
+import { Logger } from '@map-colonies/js-logger';
+import { trace, Tracer } from '@opentelemetry/api';
+import { Checksum } from '../../../src/utils/hash/checksum';
+import { ChecksumError } from '../../../src/ingestion/errors/ingestionErrors';
+import type { ChecksumProcessor } from '../../../src/utils/hash/interfaces';
+
+jest.mock('node:fs');
+jest.mock('@opentelemetry/api');
+
+describe('Checksum', () => {
+  let checksum: Checksum;
+  let mockLogger: jest.Mocked<Logger>;
+  let mockTracer: jest.Mocked<Tracer>;
+  let mockChecksumProcessor: jest.Mocked<ChecksumProcessor>;
+  let mockChecksumProcessorInit: jest.Mock;
+
+  beforeEach(() => {
+    mockLogger = {
+      debug: jest.fn(),
+      error: jest.fn(),
+    } as unknown as jest.Mocked<Logger>;
+
+    mockTracer = {} as jest.Mocked<Tracer>;
+
+    mockChecksumProcessor = {
+      algorithm: 'XXH64',
+      reset: jest.fn(),
+      update: jest.fn().mockReturnThis(),
+      digest: jest.fn(),
+    } as unknown as jest.Mocked<ChecksumProcessor>;
+
+    mockChecksumProcessorInit = jest.fn().mockResolvedValue(mockChecksumProcessor);
+
+    (trace.getActiveSpan as jest.Mock) = jest.fn().mockReturnValue({
+      updateName: jest.fn(),
+    });
+
+    checksum = new Checksum(mockLogger, mockTracer, mockChecksumProcessorInit);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe('calculate', () => {
+    it('should successfully calculate checksum for a file', async () => {
+      const filePath = '/test/path/file.txt';
+      const expectedChecksum = 'abc123def456';
+      const mockStream = new Readable();
+      mockStream._read = jest.fn();
+
+      (createReadStream as jest.Mock).mockReturnValue(mockStream);
+      // Convert hex string to bigint for the digest mock
+      mockChecksumProcessor.digest.mockReturnValue(BigInt(`0x${expectedChecksum}`));
```

Comment:
check if necessary , maybe you can just generate random hash or number if needed with `faker`